packages/compass-import-export/src/components/import-modal.tsx

```diff
@@ -0,0 +1,290 @@
+import React, { useCallback, useMemo } from 'react';
+import { connect } from 'react-redux';
+import {
+  Button,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  css,
+  spacing,
+} from '@mongodb-js/compass-components';
+import type { Document } from 'mongodb';
+
+import {
+  FINISHED_STATUSES,
+  COMPLETED_STATUSES,
+  STARTED,
+  COMPLETED,
+  COMPLETED_WITH_ERRORS,
+  CANCELED,
+  FAILED,
+  UNSPECIFIED,
+} from '../constants/process-status';
+import type { ProcessStatus } from '../constants/process-status';
+import ProgressBar from './progress-bar';
+import ImportPreview from './import-preview';
+import ImportOptions from './import-options';
+import type { AcceptedFileType } from '../constants/file-types';
+import formatNumber from '../utils/format-number';
+import {
+  startImport,
+  cancelImport,
+  selectImportFileType,
+  selectImportFileName,
+  setDelimiter,
+  setStopOnErrors,
+  setIgnoreBlanks,
+  closeImport,
+  toggleIncludeField,
+  setFieldType,
+} from '../modules/import';
+import { ImportErrorList } from './import-error-list';
+import type { RootImportState } from '../stores/import-store';
+
+/**
+ * Progress messages.
+ */
+const MESSAGES = {
+  [STARTED]: 'Importing documents...',
+  [CANCELED]: 'Import canceled',
+  [COMPLETED]: 'Import completed',
+  [COMPLETED_WITH_ERRORS]: 'Import completed with following errors:',
+  [FAILED]: 'Failed to import with the following error:',
+  [UNSPECIFIED]: '',
+} as const;
+
+const closeButtonStyles = css({
+  marginRight: spacing[2],
+});
+
+type ImportModalProps = {
```

Comment:
Wow this thing takes a lot of props. Will be super handy to have this as reference for the upcoming project.